dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -48,6 +49,106 @@ String getReceiverStateStr(const ExchangeReceiverState & s)
     }
 }
 
+// If enable_fine_grained_shuffle:
+//      Seperate chunks according to packet.stream_ids[i], then push to msg_channels[stream_id].
+// If fine grained_shuffle is disabled:
+//      Push all chunks to msg_channels[0].
+// Return true if all push succeed, otherwise return false.
+// NOTE: shared_ptr<MPPDataPacket> will be hold by all ExchangeReceiverBlockInputStream to make chunk pointer valid.
```

Comment:
Can we call `chunk.clear()` in `ExchangeReceiverBase::nextResult` after `decodeChunks` in order to release chunk's memory earlier?

Reply:
If there is an `ExchangeReceiverBlockInputStream` hanging, all chunks will not be released

Reply:
I mean that calling `Chunk(String).clear()`

Reply:
I think it's helpful for memory usage. But I don't know if it's proper way, because the memory of Chunk is in protobuf.

Reply:
maybe we can add todo or some comments about it.

Reply:
After discussion, we think the delay of memory release can happen when data is skewed to one stream, which is inevitable.